operator/v1/types_network.go

```diff
@@ -788,3 +814,38 @@ const (
 	// between pods on the cluster network.
 	IPsecModeFull IPsecMode = "Full"
 )
+
+// +kubebuilder:validation:Enum:="";"Enabled";"Disabled"
+type RouteAdvertisementsEnablement string
+
+var (
+	// RouteAdvertisementsEnabled enables route advertisements for ovn-kubernetes
+	RouteAdvertisementsEnabled RouteAdvertisementsEnablement = "Enabled"
+	// RouteAdvertisementsDisabled disables route advertisements for ovn-kubernetes
+	RouteAdvertisementsDisabled RouteAdvertisementsEnablement = "Disabled"
+)
+
+// RoutingCapabilitiesProvider is a component providing routing capabilities.
+// +kubebuilder:validation:Enum=FRR-k8s
+type RoutingCapabilitiesProvider string
+
+const (
+	// RoutingCapabilitiesProviderFRRk8s determines FRR-k8s is providing
+	// advanced routing capabilities.
+	RoutingCapabilitiesProviderFRRk8s RoutingCapabilitiesProvider = "FRR-k8s"
+)
+
+// AdditionalRoutingCapabilities describes components and relevant configuration providing
+// advanced routing capabilities.
+type AdditionalRoutingCapabilities struct {
+	// providers is a set of enabled components that provide additional routing
+	// capabilities. Valid values are:
+	// - "FRR-k8s": Enables FRR routing capabilities through the deployment of
+	// FRR-k8s.
+	// +listType=atomic
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))"
```

Comment:
Should explain that each entry in the list must be unique within the godoc for the field